src/main/java/com/metamx/emitter/core/ParametrizedUriEmitter.java

```diff
@@ -120,10 +121,30 @@ public void close() throws IOException
   @Override
   public void flush() throws IOException
   {
-    synchronized (emitters) {
-      for (Emitter emitter : emitters.values()) {
-        emitter.flush();
+    Exception thrown = null;
+    for (HttpPostEmitter httpPostEmitter : emitters.values()) {
+      try {
+        httpPostEmitter.flush();
       }
+      catch (Exception e) {
+        // If flush was interrupted, exit the loop
+        if (Thread.currentThread().isInterrupted()) {
```

Comment:
I don't think this will work either. When InterruptedException is caught it clears this flag. Implementations then wrap the exception in the RuntimeException. This will not cover such a case.